_rules/SVG-image-has-name.md

```diff
@@ -0,0 +1,201 @@
+
+Browser and assistive technology support for SVG `title` and `desc` elements is currently inconsistent. Using WAI ARIA in combination with the `img` role for non-decorative `svg` elements significantly improves accessibility browser support.
+
+## Background
+
+- [Understanding Success Criterion 1.1.1: Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
+- [SVG Accessibility API Mappings](https://www.w3.org/TR/svg-aam-1.0)
+- [WAI-ARIA Graphics Module](https://www.w3.org/TR/graphics-aria-1.0/)
+- [ARIA4: Using a WAI-ARIA role to expose the role of a user interface component](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA4)
+- [ARIA6: Using aria-label to provide labels for objects](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA6)
+- [ARIA10: Using aria-labelledby to provide a text alternative for non-text content](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA10)
+
+## Test Cases
+
+### Passed
+
+#### Pass example 1
+
+The `<svg>` element has a role of `img` which takes its accessible name from its `<title>` element.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" role="img" width="100" height="100">
+  <title>A yellow circle</title>
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Pass example 2
+
+The `<svg>` element has a role of `img` and an `aria-label` containing the accessible name.
+
+```html
+<svg xmlns="https://www.w3.org/2000/svg" role="img" width="100" height="100" aria-label="A yellow circle">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Pass example 3
+
+The `<svg>` element has the role of `graphics-document` which takes its accessible name from its' child content, the `<title>` and `<desc>` elements.
+
+```html
+
+<svg xmlns="http://www.w3.org/2000/svg" role="graphics-document">
+    <title>3 circles</title>
+    <desc>3 yellow circles with a green stroke horizontally adjacent to each other
+    </desc>
+    <g role="group">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+	<circle cx="150" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+	<circle cx="250" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+    </g>
+</svg>
+```
+
+#### Pass example 4
+
+The `<circle>` element is included in the accessibility tree and takes its' accessible name from its' child content, the `<title>` element.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg">
+	<circle role="graphics-symbol" cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow">
+	    <title>Yellow circle</title>
+	</circle>
+</svg>
+```
+
+#### Pass example 5
+
+The `svg` element and both child `<g>` elements are included in the accessibility tree with the `graphics-document` and `graphics-object` roles. All three elements have accessibile names via the `title` and `aria-label` attributes.
+
+```html
```

Comment:
Where is the yellow rectangle? :-) We never know... the yellow rect and the green don't have an a11y name - a manual validation will classify two errors in this code :-) . Suggestion of code below. Logic: the relations descriptions in g elements and the name of shapes in rect and if it is rect don't change the role of it to graphics-object. rect is announced by default by img - as a user I prefer to listen "large blue rectangle, image" than "large blue rectangle, group".
'''html
<svg xmlns="https://www.w3.org/2000/svg" width="600" height="400" viewBox="200 0 400 400" role="graphics-document">
		<title>3 shapes</title>
		<g role="graphics-object" aria-label="Blue rectangle" transform="translate(100,100)">
	        <desc>Large blue rectangle with two other shapes inside it</desc>
			<rect fill="blue" aria-label=" large blue rectangle" width="200" height="100" y="-100" />
			<g role="graphics-object" aria-label="a small yellow rectangle in a larger green rectangle" transform="translate(30,-90)">
				<rect fill="green" width="50" height="90" aria-label="the green rectangle" />
				<rect fill="yellow" x="5" y="5" width="40" height="30" aria-label="the yellow rectangle" />
				</g>
			 </g>
		</svg>
```

Reply:
@jorgeponto good point but there is nothing to say that each `<rect>` has to have an accessible name :) See the example used here https://siteimprove-wgs.atlassian.net/issues/?filter=33993

The parent `<g>` element `aria-label` describes the contents of the two child elements.

```html
<g role="graphics-object" aria-label="small yellow rectangle in larger green rectangle" transform="translate(30,-90)">
	<rect fill="green" width="50" height="90"/>
	<rect fill="yellow" x="5" y="5" width="40" height="30" />
</g>
```

Reply:
The link is private :)

Reply:
So sorry @jorgeponto wrong link, I have updated it now to https://www.w3.org/TR/graphics-aria-1.0/#graphics-object

Reply:
@Brynanders Now I saw it. I updated my comment made it at two days ago.